aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -402,6 +405,13 @@ void gemm(
     const float beta,
     float *c, int64_t ldc) {
   internal::normalize_last_dims(transa, transb, m, n, k, &lda, &ldb, &ldc);
+
+// TODO: ADI: add heuristic based on shape to dispatch to sbgemm_ vs MKLDNN
+#if AT_MKLDNN_ENABLED()
+   if (mkldnn_bf16f32_gemm(transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc)) {
```

Comment:
Since this change is not only for aarch64 but also impact x86, the major concern is the MKLDNN gemm would be slower than MKL's. Some options for you to consider:
1. Move MKL call before this one so that we do not regress x86.
2. Only apply MKLDNN gemm for aarch64 in this PR so that we do not regress x86.
3. Do more benchmarking to prove that there is no regression for x86.

Reply:
I've updated the patch @jgong5 . 
I'm guarding it with aarch64 for now.